Switch default memory format of clone operator to Preserve

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31227 Fix `.to` operator to preserve strides when `self` should be returned
* #31226 Revert "Switch default memory format of to (and similar) operators to Preserve"
* #31139 [WIP] Fix cudnn channels_last descriptio problem
* #31131 Add memory_format support to `zeros`, `ones`, `full` (still need tests)
* **#30089 Switch default memory format of clone operator to Preserve**
* #30088 Switch default memory format of to (and similar) operators to Preserve
* #30087 Switch default memory format of _like operators to Preserve

Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

VitalyFedyunin committed:
Switch default memory format of clone operator to Preserve

[ghstack-poisoned]

VitalyFedyunin committed:
rebase to master + fix tests on "Switch default memory format of clone operator to Preserve"




[ghstack-poisoned]

VitalyFedyunin committed:
fix tests code on "Switch default memory format of clone operator to Preserve"


Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

[ghstack-poisoned]

VitalyFedyunin committed:
add experimental code to check model on "Switch default memory format of clone operator to Preserve"


Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

[ghstack-poisoned]

VitalyFedyunin committed:
experiments with tag on "Switch default memory format of clone operator to Preserve"


Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

[ghstack-poisoned]

VitalyFedyunin committed:
rebase to master on "Switch default memory format of clone operator to Preserve"


Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 3e3c228:
* **1/1** broken upstream at merge base 56de885 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=56de8853dad19a050bc9ce5f89c3f7572ca8a15c&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=3e3c228b9f0d9ffa860be34c9a168fe190251c97).

### 1 failure *not* recognized by patterns:
| Job | Step |
| --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_xla_linux_xenial_py3_6_clang7_build</sup> | <sup>Build</sup> |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

VitalyFedyunin committed:
Fix flake8 on "Switch default memory format of clone operator to Preserve"


Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

[ghstack-poisoned]

VitalyFedyunin committed:
Fix flake8 on "Switch default memory format of clone operator to Preserve"


Differential Revision: [D18624985](https://our.internmc.facebook.com/intern/diff/D18624985)

[ghstack-poisoned]

ppwwyyxx commented:
This may qualify as a BC-breaking change since we have some code that implicitly assumes the result of some op is contiguous. Might be better to mention it in release notes.
